### PR TITLE
use optional env for azure openai tests

### PR DESCRIPTION
**Pull request by kristapratico**
Testing out https://github.com/Azure/azure-sdk-for-python/pull/32332 for AOAI tests

Seems like a lot of files, but most are deletions since we aren't testing v0.x anymore.

~TODO: determine whether we can remove v0 tests based on customer usage~

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3548359&view=results

### CODE REVIEW

File: sdk/openai/tests.yml [hunk 1]
```diff
@@ -3,13 +3,10 @@ trigger: none
 stages:
   - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
     parameters:
-      JobName: openai_v1
-      InjectedPackages: git+https://github.com/openai/openai-python@main
       ServiceDirectory: openai
```
Comment (scbedd) [inline sdk/openai/tests.yml:6]: Add new param: `ToxEnv`: `optional,whl`

File: tools/azure-sdk-tools/ci_tools/environment_exclusions.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def is_check_enabled(package_path: str, check: str, default: Any = True) -> bool
     if package_path == ".":
         package_path = os.getcwd()
 
-    ci_enabled = get_config_setting(package_path, "ci_enabled", default)
+    ci_enabled = get_config_setting(package_path, "ci_enabled", True)
```
Comment (kristapratico) [inline tools/azure-sdk-tools/ci_tools/environment_exclusions.py:55]: This is a bug that luckily never reared it's ugly head before finding it. We absolutely want the default for `ci_enabled` to be `True` and should not rely on whatever's passed into this function. Luckily, everywhere we use this function, we either don't pass a `default` or do pass `True`.

Comment (scbedd) [reply tools/azure-sdk-tools/ci_tools/environment_exclusions.py:55]: Good catch thanks Krista!

File: sdk/openai/tests.yml [hunk 3]
```diff
@@ -24,24 +22,3 @@ stages:
         AZURE_OPENAI_SEARCH_KEY: $(COGNITIVE-SEARCH-API-KEY)
         AZURE_OPENAI_SEARCH_INDEX: $(COGNITIVE-SEARCH-API-INDEX)
         AZURE_TEST_RUN_LIVE: 'true'
-  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
```
Comment (scbedd) [inline sdk/openai/tests.yml:27]: Side note, this simplification means that we have I have one less `tests.yml` to special case in the `1es` transition 👍 👍 👍

**Review by scbedd — APPROVED** [review]
image

Gotta love it.

I am super excited to see the `optional` env getting use!

**Pull request merged**